avocado/core/job.py

```diff
@@ -539,6 +542,9 @@ def run(self):
                  :mod:`avocado.core.exit_codes` for more information.
         """
         runtime.CURRENT_JOB = self
+        for ext in self.job_pre_post_dispatcher.extensions:
```

Comment:
This is a tough call. The plugin is called job_pre plugin, which means it should run before the job, but at this point the job results and logging is not yet set, which means we don't get any output from the plugins, unless we use an extra logger. Theoretically we should not even depend on the results dir as it is not initialized (it is created in __init__, but there are few  additional steps to properly initialize it, which happens inside the `job._run`.

So it'd be simpler to move this after results dir initialization and say the job did not started yet:

```python
diff --git a/avocado/core/job.py b/avocado/core/job.py
index d8e6b10..b372bac 100644
--- a/avocado/core/job.py
+++ b/avocado/core/job.py
@@ -463,6 +463,9 @@ class Job(object):
         """
         self._setup_job_results()
         self.__start_job_logging()
+        for ext in self.job_pre_post_dispatcher.extensions:
+            if hasattr(ext.obj, 'pre'):
+                ext.obj.pre(self)
 
         try:
             test_suite = self._make_test_suite(self.urls)
@@ -542,9 +545,6 @@ class Job(object):
                  :mod:`avocado.core.exit_codes` for more information.
         """
         runtime.CURRENT_JOB = self
-        for ext in self.job_pre_post_dispatcher.extensions:
-            if hasattr(ext.obj, 'pre'):
-                ext.obj.pre(self)
         try:
             return self._run()
         except exceptions.JobBaseException as details:
```

or we should avoid passing the uninitialized values to job-pre plugin and add additional loggers for those plugins. The problem then is, where to store these logs as the results dir is still "non-existing".